### DISCUSSION TITLE
How to debug or resolved ReactError:185?

### DISCUSSION THREAD

**Discussion by npatel-98** [vercel/next.js#70464] [Help]
### Summary

I'm working on a project in which if I directly hit a PLP url like `http://localhost:3000/categories/dog/food/dry-food`, than it gives a react error of below screenshot.
<img width="960" alt="Screenshot 2024-09-25 at 7 41 16 PM" src="https://github.com/user-attachments/assets/989b1ad6-a796-408d-af02-31f0a880b71d">

Upon searching on this, I came to know that this might be an issue where due to continuous calls, we might be stopped after a threshold. I also saw a series of `__nextjs-original-stack-frame` errors on `network` tab which I was unable to debug from which component it might be causing the issue. Can anyone help me on this on below pointers?
1. How do I debug this? From where this is being called as I cannot see a proper file in the `initiator` section on `network` tab  on the Inspect Tools.
2. How to get rid of this?

<img width="520" alt="Screenshot 2024-09-25 at 7 45 15 PM" src="https://github.com/user-attachments/assets/e3f0e558-0c0f-4245-a263-93d9976fb138">

![image](https://github.com/user-attachments/assets/5a8ed624-ff55-47f2-87de-dab0f3f1dc37)

Thanks!

### Additional information

```js
A rendering error occurred: Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops..
```

**Comment by mandorakannu** [comment]
This is happening because your react components re-render time and again. You have to look those components where you are using `useEffect()` hook or calling an API.

**Reply by npatel-98** [reply]
Yes, if I cannot get which component is exactly causing this issue, I have no other option left but to hit-and-try the different components to figure out the root cause one.

**Comment by icyJoseph** [comment]
IIRC, you could use the browser performance dev tool, make it reload-and-record, and from there, you'll probably see a gigantic stack of calls to batch update or something like that, I don't remember.

One could also start to comment out sections of the React tree, that make sup this page, until the error stops. There might be several instances of the issue.

This typically happens when you setState as you render, or have a run-away condition, that leads to setState being called over and over. 

You could look into https://github.com/kentcdodds/stop-runaway-react-effects, and make it hijack effects. 

Also, IIRC, the error stack, should contain the component that's causing the issue 🤔 - I am gonna guess it is a PopOver issue, but let's here more from you.

**Reply by npatel-98** [reply]
Thanks, but the error stack is not throwing the component name where the issue is caused. I exactly wanted this to work as it would make debugging more easy, else I'll need to comment react sections one by one and find the root cause.

**Reply by icyJoseph** [reply]
Not even expanding it? I think they (next.js) have a max number of lines, for errors 🤔 darn, but commenting sections can work, faster than you might think.

Are you using radix-ui directly, yourself? Maybe a 3rd party library contains the issue within itself?

**Reply by icyJoseph** [reply]
Also, does the console have more of the stack trace? And could you maybe add a breakpoint on `caught exceptions`? not sure that one would catch it though

**Reply by icyJoseph** [reply]
@npatel-98 did you manage to figure it out?
